packages/apputils/package.json

```diff
@@ -42,13 +42,13 @@
     "@phosphor/signaling": "^1.2.2",
     "@phosphor/virtualdom": "^1.1.2",
     "@phosphor/widgets": "^1.6.0",
-    "@types/react": "~16.0.19",
-    "react": "~16.4.0",
-    "react-dom": "~16.4.0",
+    "@types/react": "~16.4.13",
```

Comment:
This exporting of the `@types/react` package actually has pretty significant downstream effects for extension developers using react. Deduping of the typings is not particularly reliable and it can cause collisions. Changing the `@types` version is likely to do that. We should probably still keep up to date, just flagging this in case people come with issues.

Reply:
Is the intent of exporting `@types/react` here to avoid version conflicts downstream?  I ask because I am a downstream user at work and I had to add a `node` `resolution` anyway to overcome incompatibilities.  I think it would be better not to have any `@types` as dependencies.

Reply:
Yeah, the idea was to specify typings for downstream users since they are part of the *public* API of the vdom renderer. As far as I know `apputils` only package in the codebase where that is the case. There was some back-and-forth on this a few months ago, and it definitely complicates things for downstream users. I also have had to add `resolution`s to various places.

Reply:
Ugh, yeah, I see now.  We have the same problem with CodeMirror typings.  Thanks for the clarification!

Reply:
cf #4367 and #4386